### PR TITLE
New option "preserveWhiteSpace"

### PR DESCRIPTION
**Pull request by erwan-carlig**
Hello ! In our product we sometimes have to select an option based on a user-defined list of options.
These options can contain leading / trailling spaces which tagify will trim.

This PR adds a new option "preserveWhiteSpace", preventing values to be trimmed before being added as tags.

Also in this PR :
- Rename "getInvaildTagParams" to "getInvalidTagParams"
- Updated "isTagWhitelisted" documentation

Thanks in advance and have a great day !

### CODE REVIEW

**Comment by yairEO** [comment]
i will look at this tomorrow, I was away for a couple days

File: src/tagify.js [hunk 1]
```diff
@@ -772,12 +773,12 @@ Tagify.prototype = {
...
     validateTag( tagData ){
-        var value = tagData.value.trim(),
-            _s = this.settings,
+        var _s = this.settings,
+            value = _s.preserveWhiteSpace ? tagData.value : tagData.value.trim(),
             result = true;
 
         // check for empty value
-        if( !value )
+        if(_s.preserveWhiteSpace && !value || !_s.preserveWhiteSpace && !value.trim() )
```
Comment (yairEO) [inline src/tagify.js:781]: `!value.trim()` is enough here, it shouldn't matter about the `preserveWhiteSpace` setting, but only make sure there is any value at all and not multiple spaces for example

**Comment by yairEO** [comment]
Thanks for this PR! I will make some modifications and push this as part of the upcoming version (hopefully today)

**Comment by yairEO** [comment]
added in version `3.17.0`

A new setting: `trim` which is `true` by default

**Comment by erwan-carlig** [comment]
Hello !
Thanks a lot for the new setting !
Indeed a dedicated local trim method looks better ! 😄

**Pull request closed without merging**